### PR TITLE
fixed for Cygwin

### PR DESCRIPTION
**Pull request by cn00**
a little fix for compile and use distcc[d] in Cygwin

### CODE REVIEW

File: src/rpc.h [hunk 1]
```diff
@@ -49,3 +52,6 @@ int dcc_r_argv(int ifd,
                const char *argc_token,
                const char *argv_token,
                /*@out@*/ char ***argv);
+
+
+#endif
```
Comment (TafThorne) [inline src/rpc.h]: A newline at the end of the file would be nice.  I am a traditionalist.

File: src/rpc.h [hunk 2]
```diff
@@ -24,6 +24,9 @@
                 /* His hand is stretched out, and who shall turn it back?
                  * -- Isaiah 14:27 */
 
+#ifndef __DISTCC_RPC_H__
+#define __DISTCC_RPC_H__
+
```
Comment (TafThorne) [inline src/rpc.h]: I am surprised that this was not already in here.  Thank you for the addition.

File: src/exec.c [hunk 3]
```diff
@@ -61,7 +61,9 @@
 #include <sys/poll.h>
 
 #ifdef __CYGWIN__
-    #define NOGDI
+    /* #define NOGDI */
```
Comment (TafThorne) [inline src/exec.c]: If the define is not longer needed, I would rather it was removed than commented out.  I guess someone should check the behaviour of a handful of CYGWIN versions .

**Comment by TafThorne** [comment]
Over 1200 new lines, I would hate to see what a large fix was.  

Quick look and config.sub seems to have been brought up to the latest version from:  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub which seems fine to me.  We possibly should try and automate something to keep things more up to date or at least nag us if we we are behind.  So that is 611 lines that could be checked with a diff later.  

Similarly config.guess has been collected from http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess so that can be checked with a diff.  Another 1144 lines not to eyeball.  

The rest of the changes look OK to me.  


If someone could double check it all works OK on Cygwin that would be great.  If the majority of the changes are from an upstream source and we currently have a broken Cygwin compile, I am tempted to approve them and hope.  I'll give it a few days though incase anyone else has comments.

File: src/rpc.h (hunk 1 above)
Comment (cn00) [inline src/rpc.h]: Sorry for this, I can't agree with you more.

File: src/exec.c (hunk 3 above)
Comment (cn00) [inline src/exec.c]: Sorry, I prefer remove those unused code 2 years before, but my teammates changed me.

**Pull request merged**